Classes/Driver/AmazonS3Driver.php

```diff
@@ -1502,22 +1507,24 @@ protected function getSubObjects($identifier, $recursive = true, $filter = self:
      * @param array $overrideArgs
      * @return array
      */
-    protected function getListObjects($identifier, $overrideArgs = [])
+    protected function getListObjects($identifier, $overrideArgs = [], $inRecursion = false)
     {
         $args = [
             'Bucket' => $this->configuration['bucket'] ?? '',
             'Prefix' => $identifier,
         ];
         $result = $this->getCachedResponse('listObjectsV2', array_merge_recursive($args, $overrideArgs));
-
         // Cache the given meta info
         $metaInfoDownloadAdapter = GeneralUtility::makeInstance(MetaInfoDownloadAdapter::class);
-        if (isset($result['Contents']) && is_array($result['Contents'])) {
+
+        // with many files we come to the recursion which lessens the home of a cache hit, so we do not create the cache here
+        if (!$inRecursion && isset($result['Contents']) && is_array($result['Contents'])) {
```

Comment:
What's the idea of not using the cache on recursions (= large data sets, more then 1000 files)? 
IMHO the cache is urgently needed here! The cache hit should be much faster than the AWS API request. If not, the wrong cache type is used (should be in-memory here).

Reply:
As far as i remember:
Gathering those information while generating a list of object blow cache sizes, even most times a single Objects metaInfo was not accessed. (around 2 million files in one folder, it generates a huge cache und blows up ram).

Reply:
Fixed it, also in recursion it will cache the information. If a user has problems on this he can change his caching configuration which is a more complex extension configuration but at least it is possible.